### PR TITLE
added vault basic support

### PR DESCRIPTION
**Pull request by AladMocu**
Added basic support for Vault plugin

**Edited player Model:**
added balance support

**Added** `/allPlayers`  **:**
list allplayers including offline players

**Added** `/players/:uuid/pay/:value` **:**
payment `:value` to a player given `:uuid`

### CODE REVIEW

**Comment by phybros** [comment]
Thanks for your contribution @AladMocu !! Can you do me a favour and add `.classpath`, `.project`, `.settings` to `.gitignore` please?

File: src/main/java/io/servertap/api/v1/models/Player.java [hunk 1]
```diff
@@ -33,11 +33,34 @@
...
+
     public Player uuid(String uuid) {
         this.uuid = uuid;
         return this;
     }
 
+
+
+    /**
+     * Current exhaustion level
```
Comment (phybros) [inline src/main/java/io/servertap/api/v1/models/Player.java]: This comment is wrong

File: src/main/java/io/servertap/api/v1/models/OfflinePlayer.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+
+
+    /**
+     * Current exhaustion level
+     *
+     * @return exhaustion
+     **/
+    public Double getBalance() {
```
Comment (phybros) [inline src/main/java/io/servertap/api/v1/models/OfflinePlayer.java]: This is duplicated below

File: src/main/java/io/servertap/PluginEntrypoint.java [hunk 3]
```diff
@@ -48,6 +72,9 @@ public void onEnable() {
 
                 // Player routes
                 get("players", PlayerApi::playersGet);
+                get("players/:player",PlayerApi::playerGet);
+                get("allPlayers",PlayerApi::offlinePlayersGet);
+                get("players/:uuid/pay/:value",PlayerApi::playerPay);
```
Comment (phybros) [inline src/main/java/io/servertap/PluginEntrypoint.java]: Minor style thing, put a space after the comma

File: src/main/java/io/servertap/PluginEntrypoint.java [hunk 4]
```diff
@@ -5,18 +5,42 @@
...
+import org.bukkit.plugin.java.JavaPlugin;
 
 import java.util.logging.Logger;
 
 import static io.javalin.apibuilder.ApiBuilder.*;
 
 public class PluginEntrypoint extends JavaPlugin {
 
-    private final Logger log = getLogger();
+    private static final Logger log = Logger.getLogger("Minecraft");
```
Comment (phybros) [inline src/main/java/io/servertap/PluginEntrypoint.java]: Should replace this with `Bukkit.getLogger()`

File: src/main/java/io/servertap/PluginEntrypoint.java [hunk 5]
```diff
@@ -59,5 +86,53 @@ public void onEnable() {
...
+
+        if(command.getLabel().equals("test-economy")) {
+            // Lets give the player 1.05 currency (note that SOME economic plugins require rounding!)
+            sender.sendMessage(String.format("You have %s", econ.format(econ.getBalance(player.getName()))));
+            EconomyResponse r = econ.depositPlayer(player, 1.05);
+            if(r.transactionSuccess()) {
+                sender.sendMessage(String.format("You were given %s and now have %s", econ.format(r.amount), econ.format(r.balance)));
+            } else {
+                sender.sendMessage(String.format("An error occured: %s", r.errorMessage));
+            }
```
Comment (phybros) [inline src/main/java/io/servertap/PluginEntrypoint.java]: Do we need this at all? Was it left over from your testing?

**Comment by phybros** [comment]
Hi @AladMocu are you still interested in merging this?

**Comment by ATechAdventurer** [comment]
I went ahead and made some changes to this PR and opened a new one #9 . I am not trying to steal this PR I just need this endpoint for a project I am working on and wanted to push this along.

**Comment by phybros** [comment]
Closing due to the work in #9

**Pull request closed without merging**